### PR TITLE
Port custom types tests from old to new

### PR DESCRIPTION
**Pull request by pawelprazak**
# Description

On top of #777 

Part of #750 
Relates to #774

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -4,3 +4,10 @@
  [#776](https://github.com/pulumi/pulumi-java/pull/776)
 
 ### Bug Fixes
+
+- [sdk] [#774](https://github.com/pulumi/pulumi-java/issues/774) 
+  fixes deserialization regression in new codegen specific part of SDK, 
+  complex @CustomType's are deserializable now
```
Comment (t0yv0) [inline CHANGELOG_PENDING.md:10]: Seems duplicative changelog but I can tidy it up later before releasing

**Review by t0yv0 — APPROVED** [review]
This fixes the following scenario for me:


```
pulumi new kubernetes-java
# edit pom.xml to ref pulumi 0.0.1 built from source
# edit pom.xml to ref latest kubernetes
mvn clean
pulumi preview
```

This works with pre-release but not released version of Pulumi. Good job.

File: CHANGELOG_PENDING.md (hunk 1 above)
Comment (pawelprazak) [reply CHANGELOG_PENDING.md:14]: yes, there were two regressions, the second one was just a debug message

Comment (pawelprazak) [reply CHANGELOG_PENDING.md:10]: sorry, I missed it

Comment (pawelprazak) [reply CHANGELOG_PENDING.md:10]: should be ok now

**Pull request merged**